### ISSUE TITLE
Filters breaking in december!!

### ISSUE THREAD

**Issue by rammanoj** [wikimedia/WikiContrib#102]
The ``start_date`` and ``end_date`` filters of the app are breaking in december month. The reason for this might be due to the line ``month = month + 1`` in the code. It should be replace with ``month = (month + 1)%12``. There need to be some more code changes.

**Comment by Je-ni** [comment]
Hi, I'm from Outreachy. I'd like to work on this issue.

**Comment by srish** [comment]
@Je-ni hi and welcome! if you have any questions feel free to reach out in the chat: https://wikimedia.zulipchat.com/#narrow/stream/220258-gsoc20-outreachy20/topic/WikiContrib

**Comment by gotham3** [comment]
Hello! I'm an outreachy applicant and would like working on this issue!

**Comment by Je-ni** [comment]
Hi, going through issue #98 and the PR sent for #99, it seems this issue has been resolved and awaiting PR merge. Please confirm.

**Comment by srish** [comment]
@Je-ni Thanks for bringing this up. Indeed, there is a PR awaiting merge. I can't remember if there were any reasons behind not merging it. There are two options:
* You could test that PR and push any necessary changes to it.
* I can merge that PR, and then you test this issue and create a new PR if necessary. 

Which approach would you like to take?

**Comment by Je-ni** [comment]
Hi @srish, the backend is taking a bit of time to set up in my system as I don't have a python set up already. So I would prefer to go with option 2.

**Comment by srish** [comment]
@Je-ni I just realized that option 2 won't work as the PR creator deleted the repository a while ago :-/

**Comment by gotham3** [comment]
@srish Hello kindly assign me to this task I can fix it!

**Comment by srish** [comment]
@Je-ni Are you still interested in working on this task? If not, I will consider assigning it to @gotham3.

**Comment by Je-ni** [comment]
Hi @srish, please reassign it to @gotham3. I've been having `python interpreter not found` errors in my windows 10.

**Comment by shreyaa-s** [comment]
Hey @srish , I am Outrachy applicant and i would like to work on this issue. Can you please assign this to me? Thanks
PS: i have already setup the repo.

**Comment by eeshadutta** [comment]
Hi @srish Could I take up this issue?

**Comment by gotham3** [comment]
Hi @srish 
Kindly review my PR
https://github.com/wikimedia/WikiContrib/pull/145

**Comment by srish** [comment]
Thanks a lot @gotham3 for your work on this! :)

**Comment by srish** [comment]
@gotham3 This PR was causing an error, so I had to revert it. Here is the error:

```
./src/result.js
Syntax error: Identifier 'st' has already been declared (75:9)

  73 |   render = () => {
  74 |     let { start_time: st, end_time: et } = this.props.filters;
> 75 |     let st = new Date(st),
     |         ^
  76 |     st_m = st.getUTCMonth(),
  77 |     et = new Date(et),
  78 |     et_m = et.getUTCMonth();
```
Would it be possible for you to send a new PR with updated changes?

**Comment by gotham3** [comment]
@srish Done ma'am! https://github.com/wikimedia/WikiContrib/pull/153

**Comment by harshika18** [comment]
@gotham3 I tried your PR but it showed wrong range in some cases like the one shown below. The range here should be December 2018 but it shows December 2019.
![wiki_pull_error](https://user-images.githubusercontent.com/44746365/77230711-1f37dd80-6bbc-11ea-9b72-6f24d3ac9d54.png)

**Comment by harshika18** [comment]
@srish I try to correct that case [#153](https://github.com/wikimedia/WikiContrib/pull/153)

**Comment by harshika18** [comment]
@srish I try to correct that case [#157](https://github.com/wikimedia/WikiContrib/pull/157) 
![wiki_my_graph](https://user-images.githubusercontent.com/44746365/77230797-d7658600-6bbc-11ea-9ba4-00e84d2ffaab.png)

**Comment by harshika18** [comment]
It will be more appropriate if the graphs should also show the data of the specified range.
@srish can I work on this issue?

**Comment by srish** [comment]
@gotham3 thank you for your changes! @harshika18 let's chat in 157.

**Comment by rammanoj** [comment]
This is the one of the major issues the app was facing in december. I will just enquire correctly if this is fixed. Outreachy students, please don't submit any more PR. I also need to check if #157 is correct PR for it. Afair the impact of this bug has to be large on code. Let me test it. I will get back to you on this PR by this weekend (mostly).

**Comment by NdibeRaymond** [comment]
@rammanoj , #172 is concerned with displaying time range correctly and doesn't fix this issue. PR #177 along with fixing the issue of filters jumping one Month ahead, also fixes this issue

**Comment by NdibeRaymond** [comment]
@srish can this be assigned to me?

**Comment by srish** [comment]
@NdibeRaymond See https://github.com/wikimedia/WikiContrib/issues/102#issuecomment-610178021.

**Comment by NdibeRaymond** [comment]
@srish , the PR I submitted that potentially fixed this bug #117 was submitted on 28th last month, it hasn't been tested, and #172 doesn't seem to fix it
